### PR TITLE
Improve summary runtime path scoring

### PR DESCRIPTION
**Pull request by mohanagy**
## Summary by CodeRabbit

* **Improvements**
  * Improved runtime-path detection: expanded scoring components for candidate ranking, stronger penalties for utility/helper-like endpoints, deterministic hash-based tie-breaking, and filtering to avoid helper-only start/terminal pairs for more accurate graph summaries.

* **Tests**
  * Added and updated tests covering deterministic ordering, capped-path behavior, broader runtime-start heuristics (workers/jobs without metadata), and multiple runtime-path selection scenarios and edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/mohanagy/madar/pull/365?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src/runtime/graph-summary.ts [hunk 1]
```diff
@@ -266,25 +268,69 @@ function relationQualityScore(relation: unknown): number {
...
+    score += 6
+  }
+  if (node.predecessors.length === 0) {
+    score += 2
+  }
+  if (/\b(route|router|controller|page|layout|middleware|procedure|endpoint|handler)\b/.test(normalized)) {
+    score += 4
+  }
+  score += helperEndpointPenalty(normalized)
+  return score
```
[the remark this answers is not in the document]
Comment (mohanagy) [reply src/runtime/graph-summary.ts:347]: Fixed. Worker/processor-style runtime starts are now admitted at the runtime-path gate even when they have runtime predecessors, without broadening the public `entrypoints` surface. I added a regression for worker-start admission, updated the existing reverse-edge fixture to cover the newly visible worker path, and strengthened the capped tie-break test to assert the exact deterministic hash order instead of only checking `not alphabetical`.

File: src/runtime/graph-summary.ts [hunk 2]
```diff
@@ -128,6 +131,35 @@ function isExplicitEntrypoint(attributes: Record<string, unknown>): boolean {
...
+  )
+  if (/\b(worker|processor|consumer|queue consumer|event handler|job|task)\b/.test(roleText)) {
+    return true
+  }
+
+  const nodeText = normalizedRuntimeStartText(
+    attributes.label,
+    attributes.source_file,
+  )
+  return /\b(worker|processor|consumer|queue consumer|event handler)\b/.test(nodeText)
```
[the remark this answers is not in the document]
Comment (mohanagy) [reply src/runtime/graph-summary.ts]: Fixed. Camel-cased label/file fallbacks now split `ReportWorker`-style names before token matching, and metadata-less `job`/`task` starts are recognized from source-file hints. I kept `job`/`task` out of the combined label fallback because the broader matcher also promoted helper labels like `QueueRegistry.addJob` in the reverse-edge fixture, so the final fix preserves that guard while covering the missing worker/job cases.
